src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Reprocess.cs

```diff
@@ -635,13 +635,15 @@ public XmlSchema LoadSchema(string path, string baseuri)
             string correctUri = Path.GetFullPath(path);
             _output.WriteLine("Include uri: " + includeUri);
             _output.WriteLine("Correct uri: " + correctUri);
-            Stream s = new FileStream(Path.GetFullPath(path), FileMode.Open, FileAccess.Read, FileShare.Read, 1);
-            XmlReader r = XmlReader.Create(s, new XmlReaderSettings(), includeUri);
-            _output.WriteLine("Reader uri: " + r.BaseURI);
-            XmlSchema som = null;
-            using (r)
+            using (Stream s = new FileStream(Path.GetFullPath(path), FileMode.Open, FileAccess.Read, FileShare.Read, 1))
             {
-                som = XmlSchema.Read(r, new ValidationEventHandler(ValidationCallback));
+                XmlReader r = XmlReader.Create(s, new XmlReaderSettings(), includeUri);
+                _output.WriteLine("Reader uri: " + r.BaseURI);
+                XmlSchema som = null;
+                using (r)
+                {
+                    som = XmlSchema.Read(r, new ValidationEventHandler(ValidationCallback));
+                }
```

Comment:
Disposing the reader doesn't dispose the stream?

Reply:
only when you pass `CloseInput = true` to the reader setting